src/common/components/select-repository-list/index.js

```diff
@@ -202,7 +190,16 @@ function mapStateToProps(state) {
     snaps,
     repositories,
     repositoriesStatus,
-    selectRepositoriesForm
+    selectRepositoriesForm,
+    isAddingSnaps: isAddingSnaps(state),
+    isRepoSelected: (repo) => hasRepository(selectRepositoriesForm.selectedRepos, repo),
```

Comment:
Is this what selectors are really for? [The docs](http://redux.js.org/docs/recipes/ComputingDerivedData.html) state that selectors are for computing derived data from the props. This returns a function, it does not return transformed data.

This seems like a classic helper function, it could be a pure function defined after the component class and fulfil the same role.